--- a/packages/alex-2.3.5/alex.cabal
+++ b/packages/alex-2.3.5/alex.cabal
@@ -72,7 +72,7 @@ executable alex
 
   build-depends: base < 5
 
-  extensions: CPP
+  extensions: CPP, BangPatterns
   ghc-options: -Wall
   other-modules:
         AbsSyn
--- a/packages/alex-2.3.5/doc/configure.ac
+++ b/packages/alex-2.3.5/doc/configure.ac
@@ -9,5 +9,5 @@ FP_DIR_DOCBOOK_XSL([/usr/share/xml/docbook/stylesheet/nwalsh/current /usr/share/
 
 AC_PATH_PROG(DbLatexCmd,dblatex)
 
-AC_CONFIG_FILES([config.mk])
+AC_CONFIG_FILES([config.mk alex.1])
 AC_OUTPUT
--- haskell-platform-2011.2.0.1/packages/haskell-src-1.0.1.4/haskell-src.cabal~	2011-04-03 23:23:56.000000000 +0200
+++ haskell-platform-2011.2.0.1/packages/haskell-src-1.0.1.4/haskell-src.cabal	2011-09-27 20:12:10.939000793 +0200
@@ -32,8 +32,7 @@
   if flag(split-base)
     build-depends:      base >= 4 && < 5, syb, pretty, array
   else
-    build-depends:      base < 3
-  build-depends: haskell98
+    build-depends:      base < 3, haskell98
   -- The dependency on Haskell 98 is only because
   -- Happy generates a parser that imports Array
   extensions:   CPP
--- haskell-platform-2011.2.0.1/packages/text-0.11.0.6/text.cabal~	2011-04-03 23:24:18.000000000 +0200
+++ haskell-platform-2011.2.0.1/packages/text-0.11.0.6/text.cabal	2011-09-27 20:25:42.296965245 +0200
@@ -120,6 +120,7 @@
     build-depends:
       ghc-prim, base >= 4, deepseq >= 1.1.0.0
     cpp-options: -DHAVE_DEEPSEQ
+    extensions: BangPatterns
   else
     build-depends: extensible-exceptions
     extensions: PatternSignatures
--- haskell-platform-2011.2.0.1/packages/fgl-5.4.2.3/fgl.cabal~	2011-04-03 23:23:53.000000000 +0200
+++ haskell-platform-2011.2.0.1/packages/fgl-5.4.2.3/fgl.cabal	2011-09-27 21:22:47.148188662 +0200
@@ -38,5 +38,5 @@
 	Data.Graph.Inductive.Query.TransClos,
 	Data.Graph.Inductive
 build-depends:	base < 5, mtl, containers, array
-extensions: MultiParamTypeClasses, OverlappingInstances, FlexibleInstances, ScopedTypeVariables
+extensions: MultiParamTypeClasses, OverlappingInstances, FlexibleInstances, ScopedTypeVariables, BangPatterns
 build-type: Simple
--- haskell-platform-2011.2.0.1/packages/cabal-install-0.10.2/cabal-install.cabal.orig	2011-04-03 23:24:22.000000000 +0200
+++ haskell-platform-2011.2.0.1/packages/cabal-install-0.10.2/cabal-install.cabal	2011-09-27 21:53:05.311604026 +0200
@@ -93,33 +93,24 @@
         Paths_cabal_install
 
     build-depends: base     >= 2        && < 5,
-                   Cabal    >= 1.10.1   && < 1.11,
+                   Cabal    >= 1.10.1   && < 1.13,
                    filepath >= 1.0      && < 1.3,
                    network  >= 1        && < 3,
                    HTTP     >= 4000.0.2 && < 4001,
                    zlib     >= 0.4      && < 0.6,
-                   time     >= 1.1      && < 1.3
-
-    if flag(old-base)
-      build-depends: base < 3
-    else
-      build-depends: base       >= 3,
-                     process    >= 1   && < 1.1,
-                     directory  >= 1   && < 1.2,
-                     pretty     >= 1   && < 1.1,
-                     random     >= 1   && < 1.1,
-                     containers >= 0.1 && < 0.5,
-                     array      >= 0.1 && < 0.4,
-                     old-time   >= 1   && < 1.1
-
-    if flag(bytestring-in-base)
-      build-depends: base >= 2.0 && < 2.2
-    else
-      build-depends: base < 2.0 || >= 3.0, bytestring >= 0.9
+                   time     >= 1.1      && < 1.3,
+                   process    >= 1   && < 1.2,
+                   directory  >= 1   && < 1.2,
+                   pretty     >= 1   && < 1.2,
+                   random     >= 1   && < 1.1,
+                   containers >= 0.1 && < 0.5,
+                   array      >= 0.1 && < 0.4,
+                   old-time   >= 1   && < 1.1,
+                   bytestring >= 0.9
 
     if os(windows)
       build-depends: Win32 >= 2 && < 3
       cpp-options: -DWIN32
     else
-      build-depends: unix >= 1.0 && < 2.5
+      build-depends: unix >= 1.0 && < 2.6
     extensions: CPP
--- haskell-platform-2011.2.0.1/packages/haskell-platform-2011.2.0.1/haskell-platform.cabal.orig	2011-04-03 23:24:26.000000000 +0200
+++ haskell-platform-2011.2.0.1/packages/haskell-platform-2011.2.0.1/haskell-platform.cabal	2011-09-27 22:10:03.918772545 +0200
@@ -59,7 +59,7 @@
     haskell-src                 ==1.0.1.4,
     html                        ==1.0.1.2,
     HUnit                       ==1.2.2.3,
-    network                     ==2.3.0.2,
+    network                     ==2.3.0.5,
     OpenGL                      ==2.2.3.0,
     parallel                    ==3.1.0.1,
     parsec                      ==3.1.1,
@@ -68,10 +68,10 @@
     regex-compat                ==0.93.1,
     regex-posix                 ==0.94.4,
     stm                         ==2.2.0.1,
-    syb                         ==0.3,
+    syb                         ==0.3.3,
     xhtml                       ==3000.2.0.1,
     zlib                        ==0.5.3.1,
-    HTTP                        ==4000.1.1,
+    HTTP                        ==4000.1.2,
     deepseq                     ==1.1.0.2,
 
     -- 2011.1 proposals:
